### PR TITLE
This PR fixes and vastly improves the tests, if they don't pass changes will be much easier

### PR DESCRIPTION
**Pull request by sokorototo**
Branch pollution is real, so whenever we finish a PR we delete the branch, this ensures all branches are as close to `main` as possible to prevent excess merge conflicts

### CODE REVIEW

**Comment by sokorototo** [comment]
Hmm I don't have write access? Why? We are only two people after all

**Comment by tijlleenders** [comment]
> Hmm I don't have write access? Why? We are only two people after all. Or is it that I just need an approving review first

Yeah I communicated this by chat. Main is protected now. We'll integrate PRs that close issues, like any project. Thisvsetsbup the project for better collaboration.

File: test/goals.json [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
   {
-    "goalId": 1,
+    "id": 1,
     "description": "shopping",
-    "duration": 1,
-    "start": "2022-05-27T10:00:00.000Z",
-    "deadline": "2022-05-27T13:00:00.000Z"
+    "task_duration": 1,
+    "start": "June 1, 2022 12:00:00",
+    "deadline": "June 2, 2022 18:00:00"
```
Comment (tijlleenders) [inline test/goals.json:7]: I understand the format change but the date and time don't have to be changed.

File: ts/api.ts [hunk 2]
```diff
@@ -49,32 +71,41 @@ export interface Goal {
...
 }
 
-// An interface that describes a Task
+/**
+ * An interface that describes a Task
+ */
 export interface Task {
 	goal_id: number,
 	start: DateTime,
 	finish: DateTime,
```
Comment (tijlleenders) [inline ts/api.ts:80]: Shoukd be deadline

**Review by tijlleenders — CHANGES_REQUESTED** [review]
Onnour way!

test/goals.json date and times are not conform test case.

Also some "finish" instances that need renaming to deadline.

**Comment by tijlleenders** [comment]
Good to see the test passing!

The asserts still need to be added to make the test fail.

**Comment by sokorototo** [comment]
Asserts are already in place

**Comment by tijlleenders** [comment]
> Asserts are already in place

Where?
I only see one assert in the test:
image


What I need is assert the expected outcome:
Three tasks scheduled according to the expectation defined in the issue #3 

I'll write it here too:
Goal 1 => One task from 11-12
Goal 2 => One task from 10-11
Goal 3 => One task from 13-14

**Review by tijlleenders — APPROVED** [review]
Just merging this to restart implementation of preprocessor and scheduler logic.

**Pull request merged**